pkg/cmd/workspace/util/repository_wizard.go

```diff
@@ -74,13 +97,33 @@ func getRepositoryFromWizard(userGitProviders []apiclient.GitProvider, additiona
 	}
 
 	var providerRepos []apiclient.GitRepository
-	err = views_util.WithSpinner("Loading", func() error {
-		providerRepos, _, err = apiClient.GitProviderAPI.GetRepositories(ctx, providerId, namespaceId).Execute()
-		return err
-	})
+	page = 1
+	perPage = 100
+
+	for {
```

Comment:
Just noticed this.

This doesn't really achieve much in terms of pagination.
The expected scenario is that the list gets pages from the API as they come in. E.g. You switch to page 2, then you get the second page from the API. 
You could also get 100 results and, if the page size on the UI is 10, get the next page from the API on page 11.

This would require that you add a `Total` prop to the API response so you know how many total response you have.

Reply:
you mean something like
replacing
`repos, _, err = apiClient.GitProviderAPI.GetRepositories(ctx, providerId, namespaceId).Page(page).PerPage(perPage).Execute()`
With
```
repos, resp, err = apiClient.GitProviderAPI.GetRepositories(ctx, providerId, namespaceId).Execute()
totalRepos := resp.total

page:=1
perPage:=100

for page =1
   for 1 to 100 items
      providerRepos = append(providerRepos, repos...)
   page++  
```

Removing page and per_page query params ?

Reply:
No, the pagintion query params stay as they should. But you need to keep the total number of results in memory. E.g. github has 1000 repos, you get the first page of 100 and the total is 1000. Once you get to the TUI page that should display the 101st result, you then get the next page from the API

Reply:
can you provide some pseudo code ? i am confused 😕

Reply:
perhaps you want to support pagination on the TUI side here:
```
func selectRepositoryPrompt(repositories []apiclient.GitRepository, index int, choiceChan chan<- string, selectedRepos map[string]int) {
	items := []list.Item{}

	// Populate items with titles and descriptions from workspaces.
	for _, repository := range repositories {
		newItem := item[string]{id: *repository.Url, title: *repository.Name, choiceProperty: *repository.Url, desc: *repository.Url}
		items = append(items, newItem)
	}

	l := views.GetStyledSelectList(items)

```

Reply:
Figuring that out is part of the solution 😄 

I'm not sure myself how to implement proper pagination with our TUI. You'll probably need to use a model class and catch when the user goes to the next page.

What I'm talking a about in general is classic pagination logic.
You request data entries enough to show on one page. When you go to the next page, you get the next page from the API.

Reply:
alrighty, will come back to this..